#! /bin/sh /usr/share/dpatch/dpatch-run
## 11_keyboard_accelerators.dpatch by Goedson Teixeira Paixao <goedson@debian.org>
##
## DP: Implements some keyboard accelerators for common operations.

@DPATCH@
diff -urNad gpaint-0.3.1~/src/ui.c gpaint-0.3.1/src/ui.c
--- gpaint-0.3.1~/src/ui.c	2007-08-18 15:25:12.000000000 -0300
+++ gpaint-0.3.1/src/ui.c	2007-08-18 15:47:00.000000000 -0300
@@ -526,6 +526,9 @@
   gtk_widget_set_name (open_menu, "open_menu");
   gtk_widget_show (open_menu);
   gtk_container_add (GTK_CONTAINER (file_menu_item_menu), open_menu);
+  gtk_widget_add_accelerator (open_menu, "activate", accel_group,
+                              GDK_O, (GdkModifierType) GDK_CONTROL_MASK,
+                              GTK_ACCEL_VISIBLE);
 
   image26 = gtk_image_new_from_stock ("gtk-open", GTK_ICON_SIZE_MENU);
   gtk_widget_set_name (image26, "image26");
@@ -536,6 +539,9 @@
   gtk_widget_set_name (save_menu, "save_menu");
   gtk_widget_show (save_menu);
   gtk_container_add (GTK_CONTAINER (file_menu_item_menu), save_menu);
+  gtk_widget_add_accelerator (save_menu, "activate", accel_group,
+                              GDK_S, (GdkModifierType) GDK_CONTROL_MASK,
+                              GTK_ACCEL_VISIBLE);
 
   image27 = gtk_image_new_from_stock ("gtk-save", GTK_ICON_SIZE_MENU);
   gtk_widget_set_name (image27, "image27");
@@ -601,6 +607,9 @@
   gtk_widget_set_name (quit_menu_item, "quit_menu_item");
   gtk_widget_show (quit_menu_item);
   gtk_container_add (GTK_CONTAINER (file_menu_item_menu), quit_menu_item);
+  gtk_widget_add_accelerator (quit_menu_item, "activate", accel_group,
+                              GDK_Q, (GdkModifierType) GDK_CONTROL_MASK,
+                              GTK_ACCEL_VISIBLE);
 
   image32 = gtk_image_new_from_stock ("gtk-quit", GTK_ICON_SIZE_MENU);
   gtk_widget_set_name (image32, "image32");
@@ -620,6 +629,9 @@
   gtk_widget_set_name (cut_menu, "cut_menu");
   gtk_widget_show (cut_menu);
   gtk_container_add (GTK_CONTAINER (edit_menu_item_menu), cut_menu);
+  gtk_widget_add_accelerator(cut_menu, "activate", accel_group, 
+                             GDK_X, (GdkModifierType) GDK_CONTROL_MASK, 
+                             GTK_ACCEL_VISIBLE);
 
   image33 = gtk_image_new_from_stock ("gtk-cut", GTK_ICON_SIZE_MENU);
   gtk_widget_set_name (image33, "image33");
@@ -630,6 +642,9 @@
   gtk_widget_set_name (copy_menu, "copy_menu");
   gtk_widget_show (copy_menu);
   gtk_container_add (GTK_CONTAINER (edit_menu_item_menu), copy_menu);
+  gtk_widget_add_accelerator(copy_menu, "activate", accel_group, 
+                             GDK_C, (GdkModifierType) GDK_CONTROL_MASK, 
+                             GTK_ACCEL_VISIBLE);
 
   image34 = gtk_image_new_from_stock ("gtk-copy", GTK_ICON_SIZE_MENU);
   gtk_widget_set_name (image34, "image34");
@@ -640,6 +655,9 @@
   gtk_widget_set_name (paste_menu, "paste_menu");
   gtk_widget_show (paste_menu);
   gtk_container_add (GTK_CONTAINER (edit_menu_item_menu), paste_menu);
+  gtk_widget_add_accelerator(paste_menu, "activate", accel_group, 
+                             GDK_V, (GdkModifierType) GDK_CONTROL_MASK, 
+                             GTK_ACCEL_VISIBLE);
 
   image35 = gtk_image_new_from_stock ("gtk-paste", GTK_ICON_SIZE_MENU);
   gtk_widget_set_name (image35, "image35");
@@ -666,6 +684,9 @@
   gtk_widget_set_name (select_all, "select_all");
   gtk_widget_show (select_all);
   gtk_container_add (GTK_CONTAINER (edit_menu_item_menu), select_all);
+  gtk_widget_add_accelerator(select_all, "activate", accel_group, 
+                             GDK_A, (GdkModifierType) GDK_CONTROL_MASK, 
+                             GTK_ACCEL_VISIBLE);
 
   image_menu_item = gtk_menu_item_new_with_mnemonic (_("_Image"));
   gtk_widget_set_name (image_menu_item, "image_menu_item");
@@ -862,11 +883,17 @@
   gtk_widget_set_name (create_new_window, "create_new_window");
   gtk_widget_show (create_new_window);
   gtk_container_add (GTK_CONTAINER (windows_menu_item_menu), create_new_window);
+  gtk_widget_add_accelerator (create_new_window, "activate", accel_group,
+                              GDK_N, (GdkModifierType) GDK_CONTROL_MASK,
+                              GTK_ACCEL_VISIBLE);
 
   close_this_window = gtk_menu_item_new_with_mnemonic (_("_Close This Window"));
   gtk_widget_set_name (close_this_window, "close_this_window");
   gtk_widget_show (close_this_window);
   gtk_container_add (GTK_CONTAINER (windows_menu_item_menu), close_this_window);
+  gtk_widget_add_accelerator (close_this_window, "activate", accel_group,
+                              GDK_W, (GdkModifierType) GDK_CONTROL_MASK,
+                              GTK_ACCEL_VISIBLE);
 
   help_menu_item = gtk_menu_item_new_with_mnemonic (_("_Help"));
   gtk_widget_set_name (help_menu_item, "help_menu_item");
